bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -637,4 +642,24 @@ private Type getSequenceType(Type tempExprType) {
                 return PredefinedTypes.TYPE_TEXT;
         }
     }
+
+    private void initializeIteratorNextReturnType() {
+        Type childType;
+        LinkedHashSet<Type> types = children.stream().map(BValue::getType).
```

Comment:
Shall we use a for loop here, as far as I know we try to avoid `streams` when possible due to performance reasons.

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/32210/commits/7da1da152eb6749283a6b8aa00675080e736199e